src/main/resources/openapi.json

```
@@ -1,286 +1,1279 @@
+                                "examples": {
+                                    "response": {
+                                        "value": {
+                                            "error_code": 409,
+                                            "message": "A consumer instance with the specified name already exists in the Kafka Bridge."
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    },
+                    "422": {
+                        "description": "One or more consumer configuration options have invalid values.",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "$ref": "#/components/schemas/Error"
+                                },
+                                "examples": {
+                                    "response": {
+                                        "value": {
+                                            "error_code": 422,
+                                            "message": "One or more consumer configuration options have invalid values."
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            },
+            "parameters": [
+                {
+                    "name": "groupid",
+                    "in": "path",
+                    "description": "ID of the consumer group in which to create the consumer.",
+                    "required": true,
+                    "schema": {
+                        "type": "string"
+                    }
+                }
+            ]
         },
-        {
-          "name": "name",
-          "in": "path",
-          "required": true,
-          "schema": {
-            "type": "string"
-          }
-        }
-      ]
-    },
-    "/consumers/{groupid}/instances/{name}/subscription": {
-      "post": {
-        "operationId": "subscribe",
-        "responses": {
-          "200": {
-            "description": "OK"
-          }
-        }
-      },
-      "delete": {
-        "operationId": "unsubscribe",
-        "responses": {
-          "200": {
-            "description": "OK"
-          }
-        }
-      },
-      "parameters": [
-        {
-          "name": "groupid",
-          "in": "path",
-          "required": true,
-          "schema": {
-            "type": "string"
-          }
+        "/consumers/{groupid}/instances/{name}": {
+            "delete": {
+                "tags": [
+                    "Consumers"
+                ],
+                "description": "Deletes a specified consumer instance. The operation MUST be performed on the the Kafka Bridge instance that contains the consumer.",
```

```suggestion
                "description": "Deletes a specified consumer instance. The operation MUST be performed on the Kafka Bridge instance that contains the consumer.",
```

"The operation MUST be performed on the Kafka Bridge instance that contains the consumer." I'm not sure users would really understand what that's supposed to mean. What you're trying to say is that if multiple bridge instances are deployed behind a load balancer then a delete request must be routed to the same instance as the preceeding create request. So why not just say that?

---

Tbh the sentence is pretty clear to me. Btw your point about load balancer is ok. So we could use "routed" instead of "performed" :-)

---

Deletes a specified consumer instance. The operation MUST be routed to the Kafka Bridge instance that contains the consumer.